regparser/tree/depth/derive.py

```diff
@@ -121,6 +124,8 @@ def derive_depths(original_markers, additional_constraints=[]):
         if idx > 1:
             pairs = all_vars[3*(idx-2):]
             problem.addConstraint(rules.triplet_tests, pairs)
+            if not relaxed_constraints:
+                problem.addConstraint(rules.markerless_sandwich, pairs)
```

Comment:
If this rule is giving you trouble, I'd recommend either turning it into an optional rule or making it a heuristic. Setting a flag for the whole function won't be a maintainable solution.